lib/src/main/kotlin/com/lemonappdev/konsist/core/declaration/KoBaseDeclaration.kt

```diff
@@ -10,4 +12,30 @@ open class KoBaseDeclaration(private val ktElement: KtElement) : KoPsiDeclaratio
      * KoFile containing the declaration
      */
     val containingFile by lazy { KoFileDeclaration.getInstance(ktElement.containingKtFile) }
+
+    val parentDeclaration by lazy {
+        val parents = ktElement
+            .parents
+            .toList()
+
+        val parent = when {
+            parents.size > 1 -> parents[1]
+            else -> null
+        }
+
+        val name = parent
+            ?.namedUnwrappedElement
+            ?.name
+
+        name?.let {
+            containingFile
+                .declarations(includeLocal = true, includeNested = true)
+                .firstOrNull { declaration -> declaration.name == it }
```

Comment:
this is a bad idea, as multiple declarations may have the same name, especially functions, and properties. 

```kotlin
class A() {
    class Log {
         fun log()
    }
}
class B() {
    class Log {
         fun log() // parentDeclaration for this will be incorrect
    }
}
```

We need an alternative approach. I think it would be best to pass a parent instance when creating a declaration:
```kotlin
cache.getOrCreateInstance(ktClass, parent)
```

In this case `ktClass` and `parent` pair would be used as the cache key:
```kotlin
KoDeclarationCache.getOrCreateInstance {
...
val cacheKey = ktClass to parent
val declaration = get(cacheKey)
...
}

```